### PR TITLE
[#5825] add hardwallet auth method and connect screens

### LINKED ISSUE #5825: Add hardwallet auth method to set up new account flow
### Description

When creating new account, screen with available authentication methods should be shown: 

authentication method

"set up password" leads to old-style password auth flow.

"set up status hardwallet" shows following screen if NFC is ON:

hardwallet product screen 2x

And this screen when NFC is OFF:

error nfc is off

### Notes

Please refer to Figma Flows for screen stylings (they are not 100% precise now, but good enough to convert them to screens).

Flows diagram: https://www.figma.com/file/UfQjpWl1hmRchHIyY3Wvu2nW/Hardwallet-light-flows

NFC Library: https://github.com/whitedogg13/react-native-nfc-manager

<blockquote><div> Figma</div><div><strong><a href="https://www.figma.com/file/UfQjpWl1hmRchHIyY3Wvu2nW/Hardwallet-light-flows">Figma</a></strong></div><div>Created with Figma</div></blockquote>
<blockquote><div> GitHub</div><div><strong><a href="https://github.com/whitedogg13/react-native-nfc-manager">whitedogg13/react-native-nfc-manager</a></strong></div><div>React Native NFC module for both Android & iOS. Contribute to whitedogg13/react-native-nfc-manager development by creating an account on GitHub.</div></blockquote>

### PR DESCRIPTION
**Pull request by dmitryn**
fixes #5825 

* add react-native-nfc-manager to detect NFC support
* add "Choose auth method" screen https://www.figma.com/file/UfQjpWl1hmRchHIyY3Wvu2nW/Hardwallet-light-flows?node-id=259%3A29
* add "Hardwallet connect" screen https://www.figma.com/file/UfQjpWl1hmRchHIyY3Wvu2nW/Hardwallet-light-flows?node-id=258%3A594

### Review notes (optional):
Hidden behind feature flag. Stylings probably change in future.

status: ready 

screenshot 2018-09-18 19 30 40
screenshot 2018-09-18 19 30 35

<blockquote><div> Figma</div><div><strong><a href="https://www.figma.com/file/UfQjpWl1hmRchHIyY3Wvu2nW/Hardwallet-light-flows">Figma</a></strong></div><div>Created with Figma</div></blockquote>
<blockquote><div> Figma</div><div><strong><a href="https://www.figma.com/file/UfQjpWl1hmRchHIyY3Wvu2nW/Hardwallet-light-flows">Figma</a></strong></div><div>Created with Figma</div></blockquote>

### CODE REVIEW

**Review by janherich — APPROVED** [review]

File: src/status_im/events.cljs [hunk 1]
```diff
@@ -141,6 +142,11 @@
  (fn [cofx [_ result password]]
    (accounts.create/on-account-created result password false cofx)))
 
+(handlers/register-handler-fx
+ :accounts.create.ui/create-new-account
```
Comment (yenda) [inline src/status_im/events.cljs]: name the event after what happened not what will happen

File: src/status_im/events.cljs [hunk 2]
```diff
@@ -481,3 +487,25 @@
  :notifications.callback/request-notifications-permissions-denied
  (fn [cofx _]
    (accounts/show-mainnet-is-default-alert cofx)))
+
+;; hardwallet module
+
+(handlers/register-handler-fx
+ :hardwallet.callback/set-nfc-support
```
Comment (yenda) [inline src/status_im/events.cljs]: same for the 4 following events

Comment (yenda) [reply src/status_im/events.cljs]: functions names are fine they should reflect what will happen, but events must reflect what happened not what will happen

File: src/status_im/hardwallet/core.cljs [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        -default
+        isSupported
+        (then #(re-frame/dispatch [:hardwallet.callback/set-nfc-support %])))))
+
+(defn check-nfc-enabled []
+  (when platform/android?
+    (.. js-dependencies/nfc-manager
+        -default
+        isEnabled
+        (then #(re-frame/dispatch [:hardwallet.callback/set-nfc-enabled %])))))
```
Comment (yenda) [inline src/status_im/hardwallet/core.cljs]: check-nfc-enabled-success

File: src/status_im/hardwallet/core.cljs [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+            [status-im.utils.config :as config]
+            [status-im.react-native.js-dependencies :as js-dependencies]
+            [re-frame.core :as re-frame]))
+
+(defn check-nfc-support []
+  (when config/hardwallet-enabled?
+    (.. js-dependencies/nfc-manager
+        -default
+        isSupported
+        (then #(re-frame/dispatch [:hardwallet.callback/set-nfc-support %])))))
```
Comment (yenda) [inline src/status_im/hardwallet/core.cljs]: check-nfc-support-success

File: src/status_im/hardwallet/core.cljs [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+(re-frame/reg-fx
+ :hardwallet/check-nfc-support
+ check-nfc-support)
+
+(re-frame/reg-fx
+ :hardwallet/check-nfc-enabled
+ check-nfc-enabled)
+
+(re-frame/reg-fx
+ :hardwallet/go-to-settings
```
Comment (yenda) [inline src/status_im/hardwallet/core.cljs]: open-nfc-settings ?

File: src/status_im/accounts/create/core.cljs [hunk 6]
```diff
@@ -98,6 +99,11 @@
     :enter-password {:db (navigation/navigate-back (:db cofx))}
     :confirm-password (reset-account-creation cofx)))
 
+(defn navigate-to-authentication-method [{:keys [db]}]
+  (if (hardwallet/hardwallet-supported? db)
+    {:dispatch [:navigate-to :hardwallet/authentication-method]}
```
Comment (yenda) [inline src/status_im/accounts/create/core.cljs]: no need to dispatch here you can use (navigation/navigate-to ...) directly

**Review by yenda — CHANGES_REQUESTED** [review]
nice work, need a few fixes regarding events names

File: src/status_im/accounts/create/core.cljs (hunk 6 above)
Comment (dmitryn) [reply src/status_im/accounts/create/core.cljs]: @yenda not sure about this with new react-native-navigation changes. There is `navigation-interceptors` call in `:navigate-to` event handler. @rasom can you clarify if `navigate-to-cofx` can be used here directly?

File: src/status_im/events.cljs (hunk 1 above)
Comment (dmitryn) [reply src/status_im/events.cljs]: @yenda does `:accounts.create.ui/create-new-account-button-pressed` sound good to you?

Comment (yenda) [reply src/status_im/events.cljs]: sure if that is what happened

**Comment by dmitryn** [comment]
@yenda changed event names, please take a look

File: src/status_im/hardwallet/core.cljs [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+  {:db (assoc-in db [:hardwallet :nfc-enabled?] enabled?)})
+
+(defn open-nfc-settings []
+  (when platform/android?
+    (.. js-dependencies/nfc-manager
+        -default
+        goToNfcSetting)))
+
+(defn navigate-to-connect-screen []
+  {:dispatch                     [:navigate-to :hardwallet/connect]
```
Comment (yenda) [inline src/status_im/hardwallet/core.cljs]: to check with @rasom as well but I think this is also something that can be done with `(navigation/navigate-to ...)`

**Review by yenda — APPROVED** [review]
lgtm but wait for @rasom input for the `:dispatch [:navigate-to...]` events

**Pull request merged**